tools/daemon/message.c

```diff
@@ -426,6 +441,21 @@ static int proc_msg_flow(struct vma_hdr *msg_hdr, size_t size, struct sockaddr_u
 		goto err;
 	}
 
+	/* Note: special loopback logic */
+	if (NULL == peeraddr) {
+		value->if_id = sys_lo_ifindex();
+		ack = 0;
+		if (value->if_id <= 0) {
+			rc = -EFAULT;
+			goto err;
+		}
+	} else if ((VMA_MSG_FLOW_TCP_5T == data->type ||
+			VMA_MSG_FLOW_UDP_5T == data->type) &&
+			sys_iplocal(value->flow.t5.src_ip)) {
+		rc = 0;
+		goto err;
+	}
```

Comment:
Please add some comments here.
We need to discuss agian about the sys_iplocal() implementation.